README.md

```diff
@@ -1,36 +1,93 @@
-# Theia applications
+<!-- Main Header  -->
+<div align='center'>
+
+<img src="./assets/theia.svg" width="125px">
+
+## Eclipse Theia Applications - Docker & Electron
+
 [![Build Status](https://travis-ci.org/theia-ide/theia-apps.svg?branch=master)](https://travis-ci.org/theia-ide/theia-apps)
+[![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-curved)](https://github.com/theia-ide/theia-apps/labels/help%20wanted)
+[![Questions](https://img.shields.io/badge/Questions-blue.svg?style=flat-curved)](https://github.com/theia-ide/theia-apps/issues?utf8=%E2%9C%93&q=label%3Aquestion+)
+
+Collection of example cloud & desktop applications built using the [Theia](https://github.com/eclipse-theia/theia) platform.
+
+</div>
 
-[Theia](https://github.com/theia-ide/theia) is a platform to develop Cloud & Desktop IDEs with modern web technologies.
+---
 
-This repository contains example Theia applications based on published Theia extension packages.
+<br />
 
-- [Theia Docker](#theia-docker)
-  - [How to use `theiaide/theia` image?](#how-to-use-theiaidetheia-image)
-  - [Image Variants](#image-variants)
-    - [`theiaide/theia:latest`](#theiaidetheialatest)
-    - [`theiaide/theia:<version>`](#theiaidetheiaversion)
-    - [`theiaide/theia:next`](#theiaidetheianext)
-    - [`theiaide/theia:<version>-next.<commit hash>`](#theiaidetheiaversion-nextcommit-hash)
-- [Theia Desktop](#theia-desktop)
-- [Contributing](CONTRIBUTING.md)
+### Outline
+
+- [Overview](#overview)
+- [Docker Image Variants](#docker-image-variants)
+- [Additional Docker Examples](#additional-docker-examples)
+    - [How to use `theiaide/theia` image](#how-to-use-theiaidetheia-image)
+- [Electron Applications](#electron-apps)
+- [Running & Debugging](#running-&-debugging)
+- [Tips & Tricks](#tips-&-tricks)
+    - [Build Options](#build-options)
+- [Contributing](#contributing)
 - [License](#license)
 
-## Theia Docker
+<br />
+
+### Overview
+
+This repository contains a collection of Theia-based IDE applications (both electron desktop apps and docker images) as examples, for demo and continuous-integration purposes.
+
+### Docker Image Variants
+
+<img
+    src='./assets/docker.png'
+    alt='docker logo'
+    width="125px"
+/>
 
-[![dockeri.co](http://dockeri.co/image/theiaide/theia)](https://hub.docker.com/r/theiaide/theia/)
+| Image Name | Description | Documentation |
+|:---|:---|:---|
+| [theiaide/theia](https://hub.docker.com/r/theiaide/theia) | Theia-based JavaScript/TypeScript (Web Technologies) example application | |
 
-`theiaide/theia` image contains an example of Theia based IDE for Web Developers.
+_Other Variants_:
+
+| Image Name | Description | Documentation |
+|:---|:---|:---|
+| [theiaide/theia-cpp](https://hub.docker.com/r/theiaide/theia-cpp) | Theia-based C/C++ example application | [docs](./theia-cpp-docker/README.md) |
```

Comment:
It would be nice if the description was coming from the docker image as well

Reply:
> It would be nice if the description was coming from the docker image as well

@lmcbout do you mind elaborating?

Reply:
In docker, there must be a file description that could be display with the images. If the description is on the docker image, I think the info should come from there, so it would remains consistent to the image if ever that images is changed, the file description will change at the same time

Reply:
It is just a high-level description of the image for the benefit of anyone looking at the repo's readme. More information will always be available in the individual image's readme if they wish to see additional details. I want to keep a high-level description without users needing to go to each individual readme to get an understanding.